engine-transactions/src/eip_4844.rs

```diff
@@ -0,0 +1,160 @@
+use crate::eip_2930::AccessTuple;
+use crate::Error;
+use alloc::vec::Vec;
+use aurora_engine_precompiles::secp256k1::ecrecover;
+use aurora_engine_types::types::{Address, Wei};
+use aurora_engine_types::{H256, U256};
+use rlp::{Decodable, DecoderError, Encodable, Rlp, RlpStream};
+
+/// Type indicator (per EIP-2718) for access list transactions
+/// [EIP-4844 Specification](https://eips.ethereum.org/EIPS/eip-4844#specification)
+pub const BLOB_TX_TYPE: u8 = 0x03;
+
+const UNSIGNED_FIELDS: usize = 11;
+const SIGNED_FIELDS: usize = 14;
+
+/// EIP-4844 transaction kind from the CANCUN hard fork.
+///
+/// See [EIP-4844](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-4844.md)
+/// and [EIP-4844 Blob transaction](https://eips.ethereum.org/EIPS/eip-4844#blob-transaction)
+/// for more details.
+#[derive(Debug, Eq, PartialEq, Clone)]
+pub struct Transaction4844 {
+    /// ID of chain which the transaction belongs.
+    pub chain_id: u64,
+    /// A monotonically increasing transaction counter for this sender
+    pub nonce: U256,
+    /// Determined by the sender and is optional. Priority Fee is also known as Miner Tip as it is
+    /// paid directly to block producers.
+    pub max_priority_fee_per_gas: U256,
+    /// Maximum amount the sender is willing to pay to get their transaction included in a block.
+    pub max_fee_per_gas: U256,
+    /// The maximum amount of gas the sender is willing to consume on a transaction.
+    pub gas_limit: U256,
+    /// The receiving address (`None` for the zero address)
+    pub to: Option<Address>,
```

Comment:
According to EIP-4844, `to` is a required field (not allowed to be `None`). Therefore I think we should have `to: Address`. If a transaction is intended for the zero address they can explicitly specify the zero address.

Reply:
Changed: 5ba5c12abaa8356357d4f23aa216d88ef72fb39d